demos/multi_material/thermochemical_buoyancy/thermochemical_buoyancy.py

```diff
@@ -1,101 +1,102 @@
 # Thermochemical convection
-# ===========================
-#
+# ---
+
 # Rationale
-# ---------
-#
+# -
+
 # Our previous tutorial introduced multi-material simulations in G-ADOPT by
 # investigating compositional effects on buoyancy. We extend that tutorial to include
 # thermal effects, thereby simulating thermochemical convection, which is, for example,
 # essential to modelling Earth's mantle evolution.
-#
+
 # This example
-# ------------
-#
+# -
+
 # Here, we consider the entrainment of a thin, compositionally dense layer by thermal
-# convection presented in [van Keken et al. (1997)]
-# (https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/97JB01353). Inside a 2-D
-# domain heated from below, a denser material sits at the bottom boundary beneath a
-# lighter material. Whilst the compositional stratification is stable, heat transfer
-# from the boundary generates positive buoyancy in the denser material, allowing thin
-# tendrils to be entrained in the convective circulation. To resolve these tendrils
-# using the level-set approach, significant mesh refinement is needed, making the
-# simulation computationally expensive. This tutorial will be updated once the
+# convection presented in [van Keken et al. (1997)](https://doi.org/10.1029/97JB01353).
+# Inside a 2-D domain heated from below, a denser material sits at the bottom boundary
+# beneath a lighter material. Whilst the compositional stratification is stable, heat
+# transfer from the boundary generates positive buoyancy in the denser material,
+# allowing thin tendrils to be entrained in the convective circulation. To resolve these
+# tendrils using the level-set approach, significant mesh refinement is needed, making
+# the simulation computationally expensive. This tutorial will be updated soon once the
 # development of adaptive mesh refinement in Firedrake is complete. We describe below
 # the current implementation of this problem in G-ADOPT.
 
-# As with all examples, the first step is to import the `gadopt` module, which
+# As with all examples, the first step is to import the `gadopt` package, which also
 # provides access to Firedrake and associated functionality.
 
 from gadopt import *
 
-# For this problem, it is useful to define a mesh with non-uniform refinement. To this
-# end, we use the GMSH library to generate a mesh file in a format compatible with
-# Firedrake. We specifically increase vertical resolution at the top and bottom
-# boundaries of the domain.
+# For this problem, in the absence of adaptive mesh refinement, it is useful to define a
+# mesh with non-uniform spatial refinement. To this end, we use the GMSH library to
```

Comment:
I think this is the first tutorial where we use gmsh -- so perhaps a little more background to it and some context to some of the mesh generation calls below would aid a new user.